opmd_viewer/openpmd_timeseries/particle_tracker.py

```diff
@@ -0,0 +1,276 @@
+            When retrieving particles at a several iterations,
+            (for instance, with:
+            ```
+            >>> x1, = ts.get_particle( ['x'], select=pt, iteration=400 )
+            >>> x2, = ts.get_particle( ['x'], select=pt, iteration=500 )
+            ```
+            it is sometimes important that the same individual particle has
+            the same index in the array `x1` and `x2`.
+            Using `preserve_particle_index=True` ensures that this is the case.
+            However, this means that, for a particle that becomes absent at
+            a later iteration, its index in the array has to be filled also.
+            In this case, a NaN is returned at the index of this particle.
+            When `preserve_particle_index=False`, no NaN is returned (the
+            returned array is simply smaller when particles are absent) but
+            then it is not garanteed that a given particle keeps the same index
+        """
+        # Extract the particle id and sort them
+        self.selected_pid, = ts.get_particle(['id'], species=species,
+                                select=select, t=t, iteration=iteration)
+        self.selected_pid.sort()
+
+        # Register a few metadata
+        self.N_selected = len( self.selected_pid )
+        self.species = species
+        self.preserve_particle_index = preserve_particle_index
+
+    def extract_tracked_particles( self, file_handle, data_list,
+                                    species, extensions ):
+        """
+        Select the elements of each particle quantities in data_list,
+        so as to only return those that correspond to the tracked particles
+
+        Parameters
+        ----------
+        file_handle: h5py.File object
+            The HDF5 file from which to extract data
+
+        data_list: list of 1darrays
+            A list of arrays with one element per macroparticle, that represent
+            different particle quantities
+
+        species: string
+            Name of the species being requested
+
+        extensions: list of strings
+            The extensions that the current OpenPMDTimeSeries complies with
+
+        Returns
+        -------
+        A list of 1darrays that correspond to data_list, but where only the
+        particles that are tracked are kept. (NaNs may or may not be returned
+        depending on whether `preserve_particle_index` was chosen at
+        initialization)
+        """
+        # Extract the particle id, and get the extraction indices
+        pid = read_species_data(file_handle, species, 'id', extensions)
+        selected_indices = self.get_extraction_indices( pid )
+
+        # For each particle quantity, select only the tracked particles
+        for i in range(len(data_list)):
+            if len(data_list[i]) > 1:  # Do not apply selection on scalars
+                data_list[i] = self.extract_quantity(
+                    data_list[i], selected_indices )
+
+        return( data_list )
```

Comment:
note: this is [a bit superfluous](http://stackoverflow.com/a/4979486/2719194) for single argument returns, since `return` is not a function and nothing needs to be grouped.

It's a bit like writing `(a * b ) + c` so one should omit it.